app/code/Magento/CheckoutAgreementsGraphQl/Model/Resolver/DataProvider/CheckoutAgreements.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CheckoutAgreementsGraphQl\Model\Resolver\DataProvider;
+
+use Magento\CheckoutAgreements\Api\CheckoutAgreementsListInterface;
+use Magento\CheckoutAgreements\Api\Data\AgreementInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+
+/**
+ * Checkout Agreements data provider
+ */
+class CheckoutAgreements
+{
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var CheckoutAgreementsListInterface
+     */
+    private $checkoutAgreementsList;
+
+    /**
+     * @param CheckoutAgreementsListInterface $checkoutAgreementsList
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     */
+    public function __construct(
+        CheckoutAgreementsListInterface $checkoutAgreementsList,
+        SearchCriteriaBuilder $searchCriteriaBuilder
+    ) {
+        $this->checkoutAgreementsList = $checkoutAgreementsList;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+    }
+
+    /**
+     * Get All Active Checkout Agreements Data
+     *
+     * @return array
+     */
+    public function getData(): array
+    {
+        $this->searchCriteriaBuilder->addFilter(AgreementInterface::IS_ACTIVE, true);
```

Comment:
Need to add support of `store` (with tests coverage)

Also, looks using the collection (which we can consider as query API)
```
$agreements = $this->agreementCollectionFactory->create();
$agreements->addStoreFilter($this->_storeManager->getStore()->getId()); // TODO: store should be get from query context
 $agreements->addFieldToFilter('is_active', 1);
```